apps/ledger-live-desktop/src/newArch/features/ModularDrawer/Web3AppWebview/AssetAndAccountDrawer.ts

```diff
@@ -0,0 +1,79 @@
+import { Account, AccountLike } from "@ledgerhq/types-live";
+import { Observable } from "rxjs";
+import { WalletAPIAccount } from "@ledgerhq/live-common/wallet-api/types";
+import { CryptoOrTokenCurrency } from "@ledgerhq/types-cryptoassets";
+import { setDrawer } from "~/renderer/drawers/Provider";
+import SelectAssetFlow from "../components/SelectAssetFlow";
+import SelectAccountDrawer from "~/renderer/drawers/DataSelector/SelectAccountDrawer";
+
+type Result = {
+  account: AccountLike;
+  parentAccount?: Account;
+};
+
+type DrawerParams = {
+  type: "promise" | "callback";
```

Comment:
Rather than having this `type` prop why not exporting `openAssetAndAccountDrawer` with the callbacks and:
```ts
export funtion openAssetAndAccountDrawerPromise (drawerParams: Omit<DrawerParams, "onSuccess" | "onCancel">) {
  return new Promise((resolve, reject) =>
    openAssetAndAccountDrawer({
      ...drawerParams,
      onSuccess: resolve,
      onCancel: () => new Error("Canceled by user"),
    })
  ).finally(() => setDrawer())
}
```
WDYT ?